docs/cookbook.md

```diff
@@ -0,0 +1,412 @@
+```bash
+bin/console doctrine:schema:update --dump-sql
+```
+
+In our case, this will output something like this : 
+```sql
+CREATE TABLE refdata_supplier (id INT AUTO_INCREMENT NOT NULL, code VARCHAR(25) NOT NULL, name VARCHAR(255) NOT NULL, UNIQUE INDEX UNIQ_EF5B01F877153098 (code), PRIMARY KEY(id));
+```
+
+Now run the command by replacing the `--dump-sql` option by `--force` to create the table in the database :
+```bash
+bin/console doctrine:schema:update --force
+```
+
+You can check that the table is correctly created in your database.
+ 
+## Your reference data as an attribute
+Now declare the new reference data in the application : 
+```yaml
+#app/config/config.yml
+pim_reference_data:
+    supplier:
+        class: Acme\Bundle\SupplierBundle\Entity\Supplier
+        type: simple
+```
+
+To check every thing is OK so far, execute the following commands :
+```bash
+#check if the entity doctrine configuration is OK
+bin/console doctrine:mapping:info
+```
+```bash
+#check if the reference data is correct
+bin/console pim:reference-data:check
+```
+
+At this point, the reference data can already be used as an attribute in your families. Try to create a new "reference data simple select" attribute.
+
+## Reference data CRUD
+
+### Menu
+A new menu item can be added to access the reference data list. To achieve that, declare the menu item by creating a new configuration file :
+```yaml
+#Acme\Bundle/SupplierBundle/Resources/config/form_extensions/menu.yml
+extensions:
+    acme-menu-supplier:
+        module: pim/menu/item
+        parent: pim-menu-reference_data-navigation-block
+        position: 10
+        config:
+            title: acme.menu.item.reference_data.supplier
+            to: pim_customentity_index
+            routeParams:
+                customEntityName: supplier
+```
+At this point the item menu should be visible at the end of the menu. You may have to clear your cache (including the front end cache) to see it.
+
+### Datagrid configuration
+
+Create a new configuration file to declare the datagrid configuration for supplier. First, declare the source and columns configuration just to have a basic list page : 
+
+```yaml
+#Acme/Bundle/SupplierBundle/Resources/config/datagrid/supplier.yml
+datagrid:
+    supplier:
+        options:
+            entityHint: supplier
+            manageFilters: false
+        source:
+            type: pim_datasource_default
+            entity: Acme\Bundle\SupplierBundle\Entity\Supplier
+            repository_method: createDatagridQueryBuilder
+        columns:
+            code:
+                label: acme.supplier.field.code
+            name:
+                label: acme.supplier.field.name
+```
+
+#### Form extension
+
+Declare the reference data index page form extension. Exemple of working configuration : https://github.com/akeneo-labs/CustomEntityBundle/blob/master/docs/examples/CustomBundle/Resources/config/form_extensions/brand/index.yml
```

Comment:
Same here about punctuation: "no space before, 1 space after"
I let you check everywhere in this doc. I saw some others errors ;)

Reply:
Maybe at this point, you should add to comment the create button. I think it is the only one that avoids you to make work your datagrid?
If I am wrong, forget about it :)